src/app/state/store.tsx

```diff
@@ -17,26 +17,22 @@ export const middleware: Middleware[] = [
   userConsistencyCheckerMiddleware,
   notificationCheckerMiddleware,
 ];
-const defaultMiddlewareOptions = {
-  serializableCheck: process.env.NODE_ENV !== "test",
-};
-
 export const store = configureStore({
   reducer: rootReducer,
-  // [thunk, immutableStateInvariant, serializableStateInvariant] are all in the default middleware and included by default
-  // in development, with only thunk included in production.
-  // See https://redux-toolkit.js.org/api/getDefaultMiddleware#customizing-the-included-middleware
   middleware: (getDefaultMiddleware) => {
-    const newMiddleware = getDefaultMiddleware(defaultMiddlewareOptions).concat(middleware);
+    const newMiddleware = getDefaultMiddleware({
+      immutableCheck: false,
+      serializableCheck: process.env.NODE_ENV !== "test",
+    }).concat(middleware);
+
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
     if (process.env.NODE_ENV !== "production" && !window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
       newMiddleware.concat([reduxLogger]);
     }
+
     return newMiddleware;
   },
-  preloadedState: {},
-  devTools: process.env.NODE_ENV !== "production",
```

Comment:
Do we not need these?

Reply:
I think so because it should keep the preloadedState and devTools configurations as they are important for initialising the Redux store with a predefined state and enabling Redux DevTools during development. 

- middleware: Disables only the immutableCheck while keeping the serializableCheck enabled.
- preloadedState: Initialises the store with an empty state object - it can be customised if needed.
- devTools: Enables Redux DevTools only in the development environment.

Reply:
Ok but you've removed them is what I mean ;)

Reply:
- Insert facepalm emoji - that was very unintentional

Reply:
Rectified it now!